README.md
[![CRAN_Status_Badge](https://www.r-pkg.org/badges/version/parfm)](https://cran.r-project.org/package=parfm)
[![Build Status](https://travis-ci.org/Oncostat/parfm.svg?branch=master)](https://travis-ci.org/Oncostat/parfm)

# `parfm`, Parametric Frailty Models in `R`
Federico Rotolo and Marco Munda

## Description
Fits Parametric Frailty Models by maximum marginal likelihood.
             Possible baseline hazards:
                 exponential, Weibull, inverse Weibull (Fréchet),
                 Gompertz, lognormal, log-skew-normal, and loglogistic.
             Possible Frailty distributions:
                gamma, positive stable, inverse Gaussian and lognormal.
                
## Details
Frailty models are survival models for clustered or overdispersed time-to-event data.
They consist in proportional hazards Cox's models with the addition of a random effect,
  accounting for different risk levels.

When the form of the baseline hazard is somehow known in advance, the parametric estimation approach can be used advantageously.
The `parfm` package provides a wide range of parametric frailty models in `R`.
The following baseline hazard families are implemented

* exponential,

* Weibull,

* inverse Weibull (Fréchet),

* Gompertz,

* lognormal,

* log-skew-normal,

* loglogistic,

together with the frailty distributions 

* gamma,

* positive stable,

* inverse Gaussian, and 

* lognormal.

Parameter estimation is done by maximising the marginal log-likelihood,
  with right-censored and possibly left-truncated data.


## Parametrisations
### Baseline hazards
The **exponential** hazard is
  $$h(t; \lambda) = \lambda,$$
with $\lambda > 0$.

The **Weibull** hazard is
  $$h(t; \rho, \lambda) = \rho \lambda t^{\rho-1},$$
with $\rho,\lambda > 0$.

The **inverse Weibull** (or **Fréchet**) hazard is
  $$h(t; \rho, \lambda) = \frac{\rho \lambda t^{-\rho - 1}}{\exp(\lambda t^{-\rho}) - 1}$$
with $\rho, \lambda > 0$.

  $$h(t; \rho, \lambda) = \rho \lambda t^{\rho-1},$$
with $\rho,\lambda > 0$.

The **Gompertz** hazard is
  $$h(t; \gamma, \lambda) = \lambda e^{\gamma t},$$
with $\gamma,\lambda > 0$.

The **lognormal** hazard is
  $$h(t; \mu, \sigma) = 
    { \phi([log t -\mu]/\sigma)} / { \sigma t [1-\Phi([log t -\mu]/\sigma)]},$$
with $\mu\in\mathbb R$, $\sigma > 0$ and $\phi(\cdot)$ and $\Phi(\cdot)$
the density and distribution functions of a standard Normal.


The **log-skew-normal** hazard is obtained as the ratio between the density
  and the cumulative distribution function
  of a log-skew normal random variable (Azzalini, 1985),
  which has density
$$f(t; \xi, \omega, \alpha) = \frac{2}{\omega t}
    \phi\left(\frac{\log(t) - \xi}{\omega}\right)
    \Phi\left(\alpha\frac{\log(t)-\xi}{\omega}\right)$$
with $\xi \in {R}, \omega > 0, \alpha \in {R}$, and where
$\phi(\cdot)$ and $\Phi(\cdot)$ are the density and distribution functions of a standard Normal random variable.
Of note, if $alpha=0$ then the log-skew-normal boils down 
to the log-normal distribution, with $\mu=\xi$ and $\sigma=\omega$.
    
The **loglogistic** hazard is
  $$h(t; \alpha, \kappa) = 
    {exp(\alpha) \kappa t^{\kappa-1} } / {
      1 + exp(\alpha) t^{\kappa}},$$
with $\alpha\in\mathbb R$ and $\kappa>0$.

### Frailty distributions
The **gamma** frailty distribution is
$$f ( u ) = \frac{\theta^{-\frac{1}{\theta}} u^{\frac{1}{\theta} - 1} \exp \left( - u / \theta \right)}	{\Gamma ( 1 / \theta )}$$
with $\theta > 0$ and where $\Gamma(\cdot)$ is the gamma function.

The **inverse Gaussian** frailty distribution is
$$f(u) = \frac1{\sqrt{2 \pi \theta}} u^{- \frac32} \exp \left( - \frac{(u-1)^2}{2 \theta u}  \right)$$
with $\theta > 0$.

The **positive stable** frailty distribution is
$$f(u) = f(u) = - \frac1{\pi u} \sum_{k=1}^{\infty} \frac{\Gamma ( k (1 - \nu ) + 1 )}{k!} \left( - u^{ \nu - 1} \right)^{k} \sin ( ( 1 - \nu ) k \pi )$$
with $0 < \nu < 1$.

The **lognormal** frailty distribution is
$$f(u) = \frac1{\sqrt{2 \pi \theta}} u^{-1} \exp \left( - \frac{\log(u)^2}{2 \theta}  \right)$$
with $\theta > 0$.
As the Laplace tranform of the lognormal frailties does not exist in closed form,
  the saddlepoint approximation is used (Goutis and Casella, 1999).

---
## References
    
Azzalini A (1985).
 A class of distributions which includes the normal ones.
 *Scandinavian Journal of Statistics*, **12**(2):171-178.
 URL [http://www.jstor.org/stable/4615982]


Cox DR (1972).
  Regression models and life-tables.
  *Journal of the Royal Statistical Society. Series B (Methodological)*, **34**:187–220.

Duchateau L, Janssen P (2008). *The frailty model*. Springer.

Goutis C, Casella G (1999).
  Explaining the Saddlepoint Approximation.
  *The American Statistician*, **53**(3):216-224.
[http://dx.doi.org/10.1080/00031305.1999.10474463](10.1080/00031305.1999.10474463).

Munda M, Rotolo F, Legrand C (2012).
  parfm: Parametric Frailty Models in R.
  *Journal of Statistical Software*, **51**(11):1-20. 
  DOI: [10.18637/jss.v051.i11](http://dx.doi.org/10.18637/jss.v051.i11)

Wienke A (2010).
  *Frailty Models in Survival Analysis*. 
  Chapman & Hall/CRC biostatistics series. Taylor and Francis.

NEWS
Changes in Version 2.7.4 (February 2017)
  o the observe Fisher information matrix is now returned 
    as attribute ('FisherI') of the returned object

Changes in Version 2.7.3 (January 2017)
  o New baseline hazard family: inverse Weibull (Fréchet)

Changes in Version 2.7 (January 2017)

  o optimization is now based on the optimx package
  o the default optimizer is now nlminb (from the stats package)
  o fixed issue in the likelihood with lognormal frailties
  o new function coeff()


Changes in Version 2.6.0 (November 2016)

  o new baseline distribution: log-skew normal


Changes in Version 2.5.10 (September 2015)

  o added Kendall's tau for the lognormal frailty distrubution


Changes in Version 2.5 (October 2012)

  o new frailty distrubution available: lognormal


Changes in Version 2.01 (June 2012)

  o Added function anova.parfm() to compare models


Changes in Version 1.01 (March 2012)

  o Added stratification


Changes in Version 0.66-2 (February 2012)

  o Added example datasets


www/index.php
<?php

$domain=ereg_replace('[^\.]*\.(.*)$','\1',$_SERVER['HTTP_HOST']);
$group_name=ereg_replace('([^\.]*)\..*$','\1',$_SERVER['HTTP_HOST']);
$themeroot='r-forge.r-project.org/themes/rforge/';

echo '<?xml version="1.0" encoding="UTF-8"?>';
?>
<!DOCTYPE html
	PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
	"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en   ">

<head>
  <title>parfm: Parametric Frailty Models in R</title>

  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
  <meta name="author" content="F. Rotolo and M. Munda" />
  <meta name="description" content="Parametric frailty models in R" />
  <meta name="keywords" content="parametric, frailty, frailty models, survival, R, package" /> 

  <link href="http://<?php echo $themeroot; ?>styles/estilo1.css" rel="stylesheet" type="text/css" />
  <link rel="stylesheet" type="text/css" href="https://r-forge.r-project.org/themes/css/gforge.css" />
  <link rel="stylesheet" type="text/css" href="https://r-forge.r-project.org/themes/rforge/css/theme.css" />
</head>
<body>

<!-- R-Forge Logo -->
<table border="0" width="100%" cellspacing="0" cellpadding="0">
<tr><td>
<a href="http://r-forge.r-project.org/"><img src="http://<?php echo $themeroot; ?>/imagesrf/logo.png" border="0" alt="R-Forge Logo" /> </a> </td> </tr>
</table>


<!-- project title  -->

<h2>Parametric Frailty Models in R</h2>
<em>F.Rotolo, M.Munda</em>
<p><a href="http://cran.r-project.org/web/packages/parfm/index.html">Information about the current plublic release on CRAN can be found here.</a>
<br><a href="http://r-forge.r-project.org/projects/parfm/">Information about the project development can be found here.</a>
<br><a href="http://www.jstatsoft.org/v51/i11">A published paper giving more details on the theory of frailty models and on the use of the <b>parfm</b> package can be found here [<a href="#MRL12">5</a>].</a><p>

<p>Frailty models [<a href="#DJ08">1</a>, <a href="#W10">2</a>] are survival models for clustered or overdispersed time-to-event data.
  They consist in proportional hazards Cox's models [<a href="#C72">3</a>] with the addition of a random effect,
  accounting for different risk levels.

<p>When the form of the baseline hazard is somehow known in advance, the parametric estimation approach can be used advantageously.
The <b>parfm</b> package provides a wide range of parametric frailty models in <b>R</b>.
The
  <ul>
    <li>gamma,</li> 
    <li>inverse Gaussian,</li> 
    <li>positive stable and</li> 
    <li>lognormal</li>
  </ul>
  frailty distributions can be specified,
  together with five different baseline hazard families:
  <ul>
    <li>Weibull,</li>
    <li>exponential,</li>
    <li>Gompertz,</li>
    <li>lognormal,</li>
    <li>loglogistic.</li> </ul>
  
<p>Parameter estimation is done by maximising the log-likelihood,
  with right-censored and possibly left-truncated data.


<h3>Parametrisations</h3>
<h4>Baseline hazards</h4>
<p>The <B>exponential</B> model is</p>
  <p align="center"><i>h</i>(<i>t; &lambda;</i>)<i> = &lambda;</i>,</p>
  <p>with <i>&lambda;&gt;0</i>.</p>
<p>The <B>Weibull</B> model is</p>
  <p align="center"><i>h</i>(<i>t; &rho;, &lambda;</i>)<i> = &rho; &lambda; t<sup>&rho;-1</sup></i>,</p>
  <p>with <i>&rho;,&lambda;&gt;0</i>.</p>
<p>The <B>Gompertz</B> model is</p>
  <p align="center"><i>h</i>(<i>t; &gamma;, &lambda;</i>)<i> = &lambda; </i>e<i><sup>&gamma;t</sup></i>,</p>
  <p>with <i>&gamma;,&lambda;&gt;0</i>.</p>
<p>The <B>lognormal</B> model is</p>
  <p align="center"><i>h</i>(<i>t; &mu;, &sigma;</i>) = 
    {<i> &phi;</i>([log<i> t -&mu;</i>]<i>/&sigma;</i>)}<i> / </i>{<i> &sigma; t </i>[<i>1-&Phi;</i>([log<i> t -&mu;</i>]<i>/&sigma;</i>)]},</p>
  <p>with <i>&mu;&in;<strong>R</strong></i>, <i>&sigma;&gt;0</i> and <i>&phi;</i>(<i>.</i>) and <i>&Phi;</i>(<i>.</i>) the density and distribution functions of a standard Normal.</p>
<p>The <B>loglogistic</B> model is</p>
  <p align="center"><i>h</i>(<i>t; &alpha;, &kappa;</i>) = 
    {exp(<i>&alpha;</i>) <i>&kappa; t<sup>&kappa;-1</sup> </i>} <i>/</i> {
      <i>1 +</i> exp(<i>&alpha;</i>) <i>t<sup>&kappa;</sup></i>},</p>
  <p>with <i>&alpha;&in;<strong>R</strong></i> and <i>&kappa;&gt;0</i>.</p>

<h4>Frailty distributions</h4>
<p>The <B>gamma</B> distribution is</p>
  <p align="center"><i>f</i>(<i>u;, &theta;</i>) = 
    {<i> u<sup>1/&theta;-1</sup> </i>e<i><sup>-u/&theta;</sup> </i>}<i>/</i>{<i> &Gamma;</i>(<i>1/&theta;</i>)<i> &theta;<sup>1/&theta;</sup> </i>},</p>
  <p>with <i>&theta;&gt;0</i> and <i>&Gamma;</i>(<i>.</i>) the Gamma function.</p>
<p>The <B>inverse Gaussian</B> distribution is</p>
  <p align="center"><i>f</i>(<i>u; &theta;</i>) = 
    (<i>2&theta;&pi;</i>)<i><sup>-1/2</sup> u<sup>-3/2</sup> </i>exp{(<i>u-1</i>)<i><sup>2</sup> / 2u&theta; </i>},</p>
  <p>with <i>&theta;&gt;0</i>.</p>
<p>The <B>positive stable</B> distribution is</p>
  <p align="center"><i>f</i>(<i>u; &nu;</i>)<i> = 
    -</i>&Sigma;<i><sub>k=1...&#8734;</sub></i>{<i>-u<sup>-(1-&nu;)k</sup> </i>sin(<i>(1-&nu;)k&pi;</i>)<i> &Gamma;</i>(<i>(1-&nu;)k+1</i>)<i>/k!</i>}<i> / &pi;u</i>,</p>
  <p>with <i>&nu;&in;(0, 1)</i> and <i>&Gamma;</i>(<i>.</i>) the Gamma function.</p>
<p>The <B>lognormal</B> distribution is</p>
  <p align="center"><i>f</i>(<i>u; &nu;</i>)<i> = 
    (<i>2&sigma;<sup>2</sup>&pi;</i>)<i><sup>-1/2</sup> u<sup>-1</sup> </i>exp{-(log <i>u</i>)<i><sup>2</sup> / 2&sigma;<sup>2</sup> </i>},</p>
  <p>with <i>&sigma;&gt;0</i>.</p>
As the Laplace tranform of the lognormal frailties does not exist in closed form,
  the saddlepoint approximation is used [<a href="#GC99">4</a>].


<!--References--><hr>
<h3>References</h3>
<p> [<a name="DJ08">1</a>] Duchateau L., Janssen P. (2008)
  <em><a href="http://www.springer.com/statistics/life+sciences,+medicine+%26+health/book/978-0-387-72834-6">The frailty model</a></em>.
  Springer.</p>
<p> [<a name="W10">2</a>] Wienke A. (2010)
  <em><a href="http://dx.doi.org/10.1201/9781420073911">Frailty Models in Survival Analysis</a></em>.
  Chapman & Hall/CRC biostatistics series. Taylor and Francis.</p>
<p> [<a name="C72">3</a>] Cox D.R. (1972)
  <a href="http://www.jstor.org/stable/2985181">Regression models and life-tables</a>. 
  <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> 34, 187–220.</p>
<p> [<a name="GC99">4</a>] Goutis C., Casella G.  (1999)
  <a href="http://dx.doi.org/10.1080/00031305.1999.10474463">Explaining the Saddlepoint Approximation</a>.
  <em>The American Statistician</em> 53(3), 216-224.</p>
<p> [<a name="MRL12">5</a>] Munda M., Rotolo F. and Legrand C. (2012)
  <a href="http://www.jstatsoft.org/v51/i11"><b>parfm</b>: Parametric frailty models in <b>R</b></a>. <em>Journal of Statistical Software</em> 51(11).</p>
</body>
</html>
